sympy/polys/densetools.py

```diff
@@ -1431,3 +1432,68 @@ def dmp_revert(f, g, u, K):
         return dup_revert(f, g, K)
     else:
         raise MultivariatePolynomialError(f, g)
+
+
+def dup_series_reversion(f, n, K):
+    r"""
+    Computes the compositional inverse of f using Newton iteration.
+    The result is computed modulo x**n.
+
+    Algorithm
+    =========
+
+    Given $f(x) = a_1 x + a_2 x^2 + \dots$ with $a_1$ invertible.
+
+    The goal is to find $g(x)$ such that:
+    $
+    f(g(x)) = x \mod x^n
+    $
+
+    The algorithm uses iterative refinement where $g_k$ is the approximation modulo $x^k$.
+
+    Initial approximation:
+    $
+    g_1(x) = x/a_1
+    $
+
+    Iterative step for $k = 2, 3, \dots, n$:
+    $
+    \delta = f(g_{k-1}(x)) - x \mod x^k
+    $
+    $
+    g_k(x) = g_{k-1}(x) - \delta/a_1 \mod x^k
+    $
+
+    This process incrementally builds the correct coefficients of the compositional inverse.
```

Comment:
I'm not sure that the algorithm description is needed in the docstring.

It could be noted that it is not necessarily a fast algorithm for large `n`.